validation/uniqueness_test.go

```diff
@@ -0,0 +1,196 @@
+package validation
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"testing"
+
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type uniqueProperties struct {
+	Name      string
+	ShortName string
+	RPC       []string
+}
+
+type chainIDSet map[uint64]bool
+
+func (s chainIDSet) AddIfUnique(id uint64) error {
+	if s[id] {
+		return fmt.Errorf("Chain with ID %d is duplicated", id)
+	}
+	s[id] = true
+	return nil
+}
+
+type chainNameSet map[string]bool
+
+func (s chainNameSet) AddIfUnique(name string) error {
+	if s[name] {
+		return fmt.Errorf("Chain with name %s is duplicated", name)
+	}
+	s[name] = true
+	return nil
+}
+
+type chainShortNameSet map[string]bool
+
+func (s chainShortNameSet) AddIfUnique(name string) error {
+	if s[name] {
+		return fmt.Errorf("Chain with short name %s is duplicated", name)
+	}
+	s[name] = true
+	return nil
+}
+
+func getGlobalChains() (map[uint]*uniqueProperties, error) {
+	chainListUrl := "https://chainid.network/chains_mini.json"
+
+	client := http.Client{}
+
+	req, err := http.NewRequest(http.MethodGet, chainListUrl, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("User-Agent", "optimism-superchain-registry-validation")
+
+	res, getErr := client.Do(req)
+	if getErr != nil {
+		return nil, err
+	}
+
+	if res.Body != nil {
+		defer res.Body.Close()
+	}
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	type entry struct {
+		ChainId   uint     `json:"chainId"`
+		Name      string   `json:"name"`
+		ShortName string   `json:"shortName"`
+		RPC       []string `json:"rpc"`
+	}
+
+	globalChains := make([]entry, 1000)
+
+	jsonErr := json.Unmarshal(body, &globalChains)
+	if jsonErr != nil {
+		return nil, err
+	}
+
+	globalChainIds := make(map[uint]*uniqueProperties)
+	for _, chain := range globalChains {
+		if globalChainIds[chain.ChainId] != nil {
+			return nil, fmt.Errorf("Chains listed at %s have duplicate chain Id %d",
+				chainListUrl, chain.ChainId)
+		}
+		for i, url := range chain.RPC {
+			normalizedURL, err := normalizeURL(url)
+			if err != nil {
+				return nil, err
+			}
+			chain.RPC[i] = normalizedURL
+		}
+		globalChainIds[chain.ChainId] = &uniqueProperties{chain.Name, chain.ShortName, chain.RPC}
+	}
+	return globalChainIds, nil
+}
+
+func TestChainsAreGloballyUnique(t *testing.T) {
+	globalChainIds, err := getGlobalChains()
+	if err != nil {
+		t.Fatal(err)
+	}
+	localChainIds := make(chainIDSet)
+	localChainNames := make(chainNameSet)
+	localChainShortNames := make(chainShortNameSet)
+
+	isExcluded := map[uint64]bool{
+		90001:    true, // sepolia/race, known chainId collision
+		11155421: true, // oplabs devnet 0, not in upstream repo
+		11763072: true, // base devnet 0, not in upstream repo
```

Comment:
Are these tests checking the upstream repo? If not, are these last two exclusions necessary?